src/Engine/Manager/SoundManager.cpp

```diff
@@ -116,7 +116,7 @@ void SoundManager::Update(float deltaTime) {
         processedSamples = new float[numSamples] {0};
 
     Pa_WriteStream(stream, processedSamples, *numProcessedSamples);
-    delete processedSamples;
+    delete[] processedSamples;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Mismatching allocation and deallocation: processedSamples [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmismatchAllocDealloc)